src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -802,6 +807,94 @@ private int decodeTransfer(
         return progress;
     }
 
+    private int payloadOffset(
+        AmqpSectionFW amqpSection)
+    {
+        int payloadOffset = amqpSection.offset() + 3; // TODO: 3 -> constant?
+
+        if (amqpSection.kind() == VALUE)
+        {
+            AmqpValueFW value = amqpSection.amqpValue();
+            AmqpType type = value.kind();
+            switch (type)
+            {
+            case BINARY1:
+                payloadOffset = value.getAsAmqpBinary().get().offset() + 1;
+                break;
+            case BINARY4:
+                payloadOffset = value.getAsAmqpBinary().get().offset() + 4;
+                break;
+            case STRING1:
+            case STRING4:
+                payloadOffset = value.getAsAmqpString().get().offset() + value.getAsAmqpString().fieldSizeLength();
+                break;
+            }
+        }
+        return payloadOffset;
+    }
+
+    private long totalPayloadSize(
+        AmqpSectionFW amqpSection)
+    {
+        long totalPayloadSize = 0;
+        if (amqpSection.kind() == VALUE)
+        {
+            AmqpValueFW value = amqpSection.amqpValue();
+            AmqpType type = value.kind();
+            switch (type)
+            {
+            case BINARY1:
+            case BINARY4:
+                totalPayloadSize = value.getAsAmqpBinary().length();
+                break;
+            case STRING1:
+            case STRING4:
+                totalPayloadSize = value.getAsAmqpString().length();
+                break;
+            }
+        }
+        else if (amqpSection.kind() == SEQUENCE)
+        {
+            totalPayloadSize = amqpSection.amqpSequence().sizeof();
+        }
+        else if (amqpSection.kind() == DATA)
+        {
+            totalPayloadSize = amqpSection.data().sizeof();
+        }
+        return totalPayloadSize;
+    }
+
+    private long sectionHeaderSize(
+        AmqpSectionFW amqpSection)
+    {
+        long headerSize = amqpSection.limit();
```

Comment:
size and limit are different concepts, normally size would be limit - offset
suggest reworking this as `headerOffset` and `headerLimit`, then returning `headerLimit - headerOffset` at the end to get header size.